packages/form-js-viewer/src/core/ConditionChecker.js

```diff
@@ -0,0 +1,79 @@
+import { unaryTest } from 'feelin';
+import { isString } from 'min-dash';
+
+export class ConditionChecker {
+  constructor(formFieldRegistry) {
+    this._formFieldRegistry = formFieldRegistry;
+  }
+
+  /**
+   * For given data, remove properties for which conditions are not met.
+   *
+   * @param {Object<string, any>} properties
+   * @param {Object<string, any>} data
+   */
+  applyConditions(properties, data = {}) {
+    const conditions = this._getConditions();
+
+    const newProperties = { ...properties };
+    for (const { key, condition } of conditions) {
+      const shouldRemove = !this.check(condition, data);
+
+      if (shouldRemove) {
+        delete newProperties[ key ];
+      }
+    }
+
+    return newProperties;
+  }
+
+  /**
+   * Check if given condition is met.
+   *
+   * @param {string} condition
+   * @param {import('../types').Data} [data]
+   *
+   * @returns {boolean}
+   */
+  check(condition, data = {}) {
+    if (!condition) {
+      return true;
```

Comment:
Should we allow to specify this `defaultValue`? I could imagine other cases where we do not default a condition to `true`, for example: `readonly` => default `false`.

Reply:
This sounds to me more like a form field concern: it knows whether it should display or not.